flink-java/src/main/java/org/apache/flink/api/java/typeutils/ObjectArrayTypeInfo.java

```diff
@@ -143,6 +143,18 @@ else if (type instanceof Class<?> && ((Class<?>) type).isArray()
 		throw new InvalidTypesException("The given type is not a valid object array.");
 	}
 
+	/**
+	 * Creates a new {@link org.apache.flink.api.java.typeutils.ObjectArrayTypeInfo} from a
+	 * {@link TypeInformation} for the component type.
+	 *
+	 * <p>
+	 * This must be used in cases where the complete type of the array is not available as a
+	 * {@link java.lang.reflect.Type} or {@link java.lang.Class}.
+	 */
+	public static <T, C> ObjectArrayTypeInfo<T, C> getInfoFor(TypeInformation<C> componentInfo) {
+		return new ObjectArrayTypeInfo<T, C>(Object[].class, componentInfo.getTypeClass(), componentInfo);
```

Comment:
Is `Object[].class` actually correct? Why not give the class of the real array type? There are a few ways to get that, for example `Array.newInstance(clazz, 0).getClass();`